src/main/InternalEnforcer.lua

```diff
@@ -371,6 +371,50 @@ function InternalEnforcer:removeFilteredPolicy(sec, ptype, fieldIndex, fieldValu
     return true
 end
 
+function InternalEnforcer:updateFilteredPolicies(sec, ptype, newRules, fieldIndex, fieldValues)
+    if fieldValues == nil or #fieldValues == 0 then
+        return false
+    end
+
+    local oldRules={}
```

Comment:
I think this should be:
```lua
local oldRules = self.model:getFilteredPolicy(sec, ptype, fieldIndex, fieldValues)
```

Since you are currently passing an empty array and instead need to first get the rules matching the filter.